File: 001/pseudoinverse/pseudoinverse.py
import time
import numpy as np

avg = []
for i in range(10):
    start_time = time.time()

    x = np.array([[0,1],[1,1],[2,1],[3,1]])
    x_inverse = np.linalg.pinv(x)
    y = np.array([[0],[1],[4],[9]])
    beta = np.dot(x_inverse, y)

    avg.append(time.time() - start_time)

print('Python:', sum(avg) / 10)

File: 001/pseudoinverse/pseudoinverse.jl
using LinearAlgebra

avg = []
for i = 1:10
    start_time = time_ns()

    x = [0 1; 1 1; 2 1; 3 1;]
    x_inverse = pinv(x)
    y = [0; 1; 4; 9]
    beta = x_inverse * y
    
    append!(avg, (time_ns() - start_time))
end

println("Julia: ", sum(avg) / 1000000000 / 10)